### PR TITLE
feat(@fepack/react-image): initialization

### PR DESCRIPTION
**Pull request by manudeli**
## @fepack/react-image initialization

For this PR, I hope to get approval quickly and then upload it to npm v0.0.0 and change it to future PRs

- Load
- useLoad

```tsx
const Example = () => (
  <>
    <ErrorBoundary fallback={() => <>load image error</>}>
      <Suspense>
        <Load src="https://ik.imagekit.io/ikmedia/women-dress-2.jpg">
          {(loaded) => <img src={loaded.src} />}
        </Load>
      </Suspense>
    </ErrorBoundary>
    <ErrorBoundary
      fallback={() => (
        <Load src="https://ik.imagekit.io/ikmedia/women-dress-2.jpg">
          {(loaded) => <img src={loaded.src} />}
        </Load>
      )}
    >
      <Suspense>
        <Load src="https://ik.imagekit.io/ikmedia/women-dress-2.webp"> // if webp make onerror this Load will throw error to ErrorBoundary
          {(loaded) => <img src={loaded.src} />}
        </Load>
      </Suspense>
    </ErrorBoundary>
  </>
);


```

### CODE REVIEW

File: websites/demo-react-image-next/src/app/layout.tsx [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+export const metadata: Metadata = {
+  title: "Suspensive - Visualization",
+};
+
+export default function RootLayout({
+  children,
+}: {
+  children: React.ReactNode;
+}) {
```
Comment (tooooo1) [inline websites/demo-react-image-next/src/app/layout.tsx]:
```suggestion
import { PropsWithChildren } from "react";

export const metadata: Metadata = {
  title: "Suspensive - Visualization",
};

export default function RootLayout({ children }: PropsWithChildren) {
```

`PropsWithChildren` is used elsewhere, so a review of the uniformity part.

File: websites/demo-react-image-next/package.json [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    "dev": "next dev --port 4001",
+    "lint": "next lint",
+    "start": "next start",
+    "type:check": "tsc --noEmit"
+  },
+  "dependencies": {
+    "@emotion/react": "^11.10.8",
+    "@emotion/styled": "^11.10.8",
+    "@jsxcss/emotion": "^1.3.9",
+    "@suspensive/react": "1.14.1",
```
Comment (tooooo1) [inline websites/demo-react-image-next/package.json:16]: 👍🏻

File: packages/react-image/src/Load.ts
Comment (tooooo1) [inline packages/react-image/src/Load.ts:1]: Some parts use `src: string;` and some parts use `src: HTMLImageElement["src"];`. Is there a reason to use it differently?

File: websites/demo-react-image-next/src/app/page.tsx [hunk 3]
```diff
@@ -0,0 +1,27 @@
+"use client";
+
+import { Load } from "@fepack/react-image";
+import { Suspense, ErrorBoundary } from "@suspensive/react";
+import Link from "next/link";
+
+export default function Home() {
+  return (
+    <div>
+      <Link href={"/load"}>to load</Link>
```
Comment (tooooo1) [inline websites/demo-react-image-next/src/app/page.tsx]:
```suggestion
      <Link href="/load">to load</Link>
```

File: websites/demo-react-image-next/next.config.js [hunk 4]
```diff
@@ -0,0 +1,7 @@
+/** @type {import('next').NextConfig} */
+const nextConfig = {
+  reactStrictMode: true,
+  transpilePackages: ['@suspensive/react', '@suspensive/react-query'],
+}
+
+module.exports = nextConfig
```
Comment (tooooo1) [inline websites/demo-react-image-next/next.config.js]:
```suggestion
/** @type {import('next').NextConfig} */
const nextConfig = {
  reactStrictMode: true,
  transpilePackages: ["@suspensive/react", "@suspensive/react-query"],
};

module.exports = nextConfig;
```

formatting

File: websites/demo-react-image-next/src/app/providers.tsx [hunk 5]
```diff
@@ -0,0 +1,23 @@
+"use client";
+
+import { MediaQueryProvider } from "@jsxcss/emotion";
+import { Suspensive, SuspensiveProvider } from "@suspensive/react";
+import { PropsWithChildren } from "react";
```
Comment (tooooo1) [inline websites/demo-react-image-next/src/app/providers.tsx]: In the case of `type`, I would like to specify it, but it seems that **Lint** is starting to rely on `@suspensive/react`. How to solve this?

**Review by tooooo1 — CHANGES_REQUESTED** [review]

**Comment by manudeli** [comment]
@tooooo1 I reflect your opinion in [bb729a9](https://github.com/fepack/image/pull/13/commits/bb729a967adb050152fcf20deeb5d78c9108ba95) with some clarifying. check it please

**Review by tooooo1 — APPROVED** [review]

**Pull request merged**